src/components/progress-bar/progress-bar.ts

```diff
@@ -0,0 +1,132 @@
+import {
+    Component,
+    ChangeDetectionStrategy,
+    HostBinding,
+    Input,
+} from 'angular2/core';
+import {isPresent, CONST} from 'angular2/src/facade/lang';
+import {OneOf} from '../../core/annotations/one-of';
+
+
+// TODO(josephperrott): Benchpress tests.
+// TODO(josephperrott): Add ARIA attributes for progressbar "for".
+
+
+/** Display modes of Progress Bar*/
+@CONST()
+class ProgressMode {
+  @CONST() static DETERMINATE = 'determinate';
+  @CONST() static INDETERMINATE = 'indeterminate';
+  @CONST() static BUFFER = 'buffer';
+  @CONST() static QUERY = 'query';
+}
+
+
+/**
+ * <md-progress-bar> component.
+ */
+@Component({
+  selector: 'md-progress-bar',
+  host: {
+    'role': 'progressbar',
+    'aria-valuemin': '0',
+    'aria-valuemax': '100',
+  },
+  templateUrl: './components/progress-bar/progress-bar.html',
+  styleUrls: ['./components/progress-bar/progress-bar.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MdProgressBar {
+  /**
```

Comment:
That's not not really a distinction I make when reading comments. I think that single-lining description blocks like this when possible does make the code more readable, since the comment tokens themselves (and the IDE color) cue your brain what's happening. (vs `{ }` on the same line where your brain has to do more work). This is just bikeshedding, though- not super important.